Examples/MAX32660/SPIMSS_DMA/main.c

```diff
@@ -0,0 +1,208 @@
+
+void spimss_callback(mxc_spimss_req_t *req, int error_code)
+{
+    if (req->deass) {
+        SPIMSS->mode |= (SPIMSS_ENABLE_SS); // Set Slave Select to HIGH.
+    }
+    SPIMSS->ctrl &= ~SPIMSS_ENABLE_TRANSACTION; // Disable SPIMSS transaction.
+
+    SPIMSS->dma &= ~(1 << 31); // Disable SPIMSS RX DMA requests
+    SPIMSS->dma &= ~(1 << 15); // Disable SPIMSS TX DMA requests
+
+    // Setting the dma flags to 0 for the next transaction.
+    dma_tx_completed = 0;
+    dma_rx_completed = 0;
+}
+
+void DMA0_IRQHandler(void)
+{
+    //Disable channel 0 interrupts.
+    mxc_dma_ch_regs_t *tx_channel_reg = MXC_DMA_GetCHRegs(0);
+    MXC_DMA_ChannelClearFlags(0, tx_channel_reg->stat);
+    MXC_DMA_SetChannelInterruptEn(0, false, false);
+    MXC_DMA_ReleaseChannel(0);
+    dma_tx_completed = 1;
+}
+
+void DMA1_IRQHandler(void)
+{
+    //Disable channel 1 interrupts.
+    mxc_dma_ch_regs_t *tx_channel_reg = MXC_DMA_GetCHRegs(1);
+    MXC_DMA_ChannelClearFlags(1, tx_channel_reg->stat);
+    MXC_DMA_SetChannelInterruptEn(1, false, false);
+    MXC_DMA_ReleaseChannel(1);
+    dma_rx_completed = 1;
+}
+
+int main()
+{
+    int i = 0;
+    mxc_spimss_req_t spimss_dma_req;
+
+    printf("\n\n************** SPIMSS-DMA Master Demo ****************\n");
+    printf("This example configures the SPIMSS to send data between the MISO (P0.10) and\n");
+    printf("MOSI (P0.11) pins over dma channel 0 and dma channel 1.  Connect these two pins\n");
+    printf("together. This demo shows 1024 byte data transfer for 100 times using dma.\n");
+    printf("During this demo you may see junk data printed to the serial port because the\n");
+    printf("console UART shares the same pins as the SPIMSS. Dma channel 0 is used as tx from\n");
+    printf("memory to SPIMSS TX FIFO and dma channel 1 is used for reading data from SPIMSS\n");
+    printf("RX FIFO to memory.\n\n");
+
+    // Preparation of the transmission data and receive data buffer.
+    for (i = 0; i < NUMBER_OF_UNIT_SIZE; i++) {
+        tx_data[i] = DATA_VALUE + i;
+        rx_data[i] = 0;
+    }
+
+    // Initialize SPIMSS with MAP_A.
+    // For MAP_A P0.10:MISO, P0.11:MOSI, P0.12:SCK, P0.13:SSEL
+    if (0 != MXC_SPIMSS_Init(SPIMSS, 0, SPI_SPEED, MAP_A)) {
+        return -1;
+    }
+
+    // Prepare the SPIMSS data transaction request.
+    spimss_dma_req.tx_data = tx_data;
+    spimss_dma_req.rx_data = rx_data;
+    spimss_dma_req.len = NUMBER_OF_UNIT_SIZE;
+    spimss_dma_req.bits = 16;
+    spimss_dma_req.deass = 1;
+    spimss_dma_req.tx_num = 0;
+    spimss_dma_req.rx_num = 0;
+    spimss_dma_req.callback = spimss_callback;
+
+    /**
+     * DMA channel configuration. There is only one instance of DMA in MAX32660.
+     * This instance has 4 channels. We want to use channel 0 and channel 1 for
+     * data tx and rx respectively.
+    */
+
+    // Releasing channel 0 an 1 to use as tx and rx channels respectively.
+    MXC_DMA_ReleaseChannel(0);
+    MXC_DMA_ReleaseChannel(1);
```

Comment:
This could be a problematic requirement when using the functions outside of the example code.  See above

Reply:
Hi @Jake-Carter , I added a structural design similar to #763. Now, it is possible to determine the DMA channels automatically or by using MXC_SPIMSS_SetRXDMAChannel and MXC_SPIMSS_SetTXDMAChannel functions. I tested the example application on MAX32660 EVSYS side and it seems working well. Can you please check again?